src/server/game/Maps/Map.cpp

```diff
@@ -1555,12 +1555,19 @@ uint16 GridMap::getArea(float x, float y) const
 {
     if (!_areaMap)
         return _gridArea;
-
-    x = 16 * (32 - x / SIZE_OF_GRIDS);
-    y = 16 * (32 - y / SIZE_OF_GRIDS);
-    int lx = (int)x & 15;
-    int ly = (int)y & 15;
-    return _areaMap[lx * 16 + ly];
+    // Correcting crash in blank lx and ly is not null, what goes out of range because it is not compiled in debug
+    try
```

Comment:
?? Why is this even here?

Reply:
Its a basic try and catch.
The try statement allows you to define a block of code to be tested for errors while it is being executed.
The catch statement allows you to define a block of code to be executed, if an error occurs in the try block.
Try and catch is always in pairs.

Reply:
Yes, I know what is the purpose. I don't know why it is here. What errors could it be here?

Reply:
Out of range values leads to you falling thru the terrain.